### PR TITLE
Plumbing most of the badging story for new tlfs

### PR DESCRIPTION
**Pull request by chrisnojima**
We're still waiting for core to give us the incoming gregor msg which triggers a refresh so that's mocked out. We're also waiting for updates to the getFavorites rpc so we get 'new' folders so that's mocked out in dev to randomly choose some folders to be new (see the actions/favorites)

What this does is
- [x] have code to refresh favorites when we get an incoming msg
- [x] switch tray icons based on redux state in a new substore for notifications
- [x] fixes badging in the widget
- [x] collapses route logging in console

### CODE REVIEW

File: desktop/app/menu-bar.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 import hotPath from '../hot-path'
 import menubar from 'menubar'
 
-const isWhite = ['linux', 'win32'].indexOf(process.platform) !== -1 ? '_white' : ''
-const menubarIconPath = resolveImage('menubarIcon', `topbar_iconTemplate${isWhite}.png`)
-const menubarLoadingIconPath = resolveImage('menubarIcon', `topbar_icon_loadingTemplate${isWhite}.png`)
+let color = 'white'
+
+if (process.platform === 'darwin') {
+  color = (systemPreferences && systemPreferences.isDarkMode()) ? 'white' : 'black'
```
Comment (chrisnojima) [inline desktop/app/menu-bar.js]: new electron supports this, so this should just work

File: shared/actions/favorite.js [hunk 2]
```diff
@@ -2,18 +2,84 @@
...
+import {badgeApp} from './notifications'
+import {canonicalizeUsernames, parseFolderNameToUsers} from '../util/kbfs'
+import _ from 'lodash'
 import type {Folder, favorite_favoriteList_rpc} from '../constants/types/flow-types'
 import type {Dispatch} from '../constants/types/flux'
-
 import type {FavoriteList} from '../constants/favorite'
+import type {Props as FolderProps} from '../folders/render'
+
+const folderToProps = (folders: Array<Folder>, username: string = ''): FolderProps => { // eslint-disable-line space-infix-ops
```
Comment (chrisnojima) [inline shared/actions/favorite.js]: moved this from the reducer into the action so we can also dispatch the badging call

File: shared/actions/favorite.js [hunk 3]
```diff
@@ -2,18 +2,84 @@
...
+        username: u,
+        you: u === username,
+        broken: false
+      }))
+    const sortName = users.map(u => u.username).join(' ')
+
+    const groupAvatar = f.private ? (users.length > 2) : (users.length > 1)
+    const userAvatar = groupAvatar ? null : users[users.length - 1].username
+    const meta = null
+    // const meta = (__DEV__ && Math.random() < 0.2) ? 'new' : null // uncomment to test seeing new before we integrate fully
```
Comment (chrisnojima) [inline shared/actions/favorite.js]: if you want to see the badging working you can replace the lines above

File: shared/folders/row.desktop.js [hunk 4]
```diff
@@ -33,7 +33,7 @@ const RowMeta = ({ignored, meta, styles}) => {
   }
 
   const metaBGColors = {
-    'new': globalColors.blue2,
+    'new': globalColors.orange,
```
Comment (chrisnojima) [inline shared/folders/row.desktop.js]: talked to design. the badge colors need to match the reason

File: shared/reducers/index.js [hunk 5]
```diff
@@ -1,29 +1,28 @@
...
 import {combineReducers} from 'redux'
-import login from './login'
-import devices from './devices'
-import search from './search'
-import profile from './profile'
-import config from './config'
-import tabbedRouter from './tabbed-router'
 import {List} from 'immutable'
 import type {State} from '../constants/reducer'
+import * as Constants from '../constants/dev'
```
Comment (chrisnojima) [inline shared/reducers/index.js]: sorting and dividing this list to clean up

File: shared/reducers/index.js [hunk 6]
```diff
@@ -55,7 +54,8 @@ const combinedReducer = combineReducers({
   updateConfirm,
   updatePaused,
   signup,
-  unlockFolders
+  unlockFolders,
+  notifications
```
Comment (chrisnojima) [inline shared/reducers/index.js]: new substore to hold this. we already had actions but no reducer

File: shared/router/meta-navigator.js [hunk 7]
```diff
@@ -112,11 +112,12 @@ class MetaNavigator extends Component {
     }
 
     if (printRoutes) {
-      console.log('%cRoute', 'font-size: x-large')
+      console.groupCollapsed && console.groupCollapsed(`Routing: ${rootComponent.displayName}/${JSON.stringify(uri.toJS())}`)
```
Comment (chrisnojima) [inline shared/router/meta-navigator.js]: wrapping this so its less noisy in the logs

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Menubar-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464977417526-7c261751cb23-7ad7e546c002/Menubar-Normal-7c261751cb23.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464977417526-7c261751cb23-7ad7e546c002/Menubar-Normal-7ad7e546c002.png)  
   Menubar-Normal comparison
 * Changed: **Folders_TLF-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464977417526-7c261751cb23-7ad7e546c002/Folders_TLF-Normal-7c261751cb23.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464977417526-7c261751cb23-7ad7e546c002/Folders_TLF-Normal-7ad7e546c002.png)  
   Folders_TLF-Normal comparison

**Comment by chrisnojima** [comment]
color change is expected

File: shared/constants/notifications.js [hunk 8]
```diff
@@ -5,5 +5,9 @@ import type {TypedAction} from '../constants/types/flux'
 // Actions
 export const log = 'notifications:log'
 export type LogAction = TypedAction<'notifications:log', {level: LogLevel, text: string}, void>
+export const badgeApp = 'notifications:badgeApp'
 
-export type NotificationAction = LogAction
+export type NotificationKeys = 'newTLFs'
+export type BadgeAppAction = TypedAction<'notifications:badgeApp', {key: NotificationKeys, on: boolean}, void>
```
Comment (chromakode) [inline shared/constants/notifications.js]: Can you reuse the `badgeApp` constant here?

File: shared/actions/favorite.js (hunk 3 above)
Comment (chromakode) [inline shared/actions/favorite.js]: Perhaps annotate this comment with "TODO" to match the other temporary comment below?

File: desktop/app/menu-bar.js [hunk 9]
```diff
@@ -15,16 +21,20 @@ export default function () {
...
     resizable: false,
     preloadWindow: true,
-    icon: menubarIconPath,
+    icon: icon,
     showDockIcon: true // This causes menubar to not touch dock icon, yeah it's weird
   })
 
   ipcMain.on('showTrayLoading', () => {
-    mb.tray.setImage(menubarLoadingIconPath)
+    mb.tray.setImage(loadingIcon)
```
Comment (chromakode) [inline desktop/app/menu-bar.js]: Does this ever actually get called?

**Comment by chromakode** [comment]
:+1:

File: shared/constants/notifications.js (hunk 8 above)
Comment (chrisnojima) [inline shared/constants/notifications.js]: no, cause flow doesn't understand the constants

File: desktop/app/menu-bar.js (hunk 9 above)
Comment (chrisnojima) [inline desktop/app/menu-bar.js]: its not but we have the icon and we'll likely add this in the future

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Menubar-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464980143128-1e2e54dc0987-046a642f3385/Menubar-Normal-1e2e54dc0987.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464980143128-1e2e54dc0987-046a642f3385/Menubar-Normal-046a642f3385.png)  
   Menubar-Normal comparison
 * Changed: **Folders_TLF-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1464980143128-1e2e54dc0987-046a642f3385/Folders_TLF-Normal-1e2e54dc0987.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1464980143128-1e2e54dc0987-046a642f3385/Folders_TLF-Normal-046a642f3385.png)  
   Folders_TLF-Normal comparison

**Comment by cjb** [comment]
@chrisnojima CI's green on this one now, okay to merge.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Menubar-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1465227074687-7a986158cf3c-50b77eb9c476/Menubar-Normal-7a986158cf3c.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1465227074687-7a986158cf3c-50b77eb9c476/Menubar-Normal-50b77eb9c476.png)  
   Menubar-Normal comparison
 * Changed: **Folders_TLF-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1465227074687-7a986158cf3c-50b77eb9c476/Folders_TLF-Normal-7a986158cf3c.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1465227074687-7a986158cf3c-50b77eb9c476/Folders_TLF-Normal-50b77eb9c476.png)  
   Folders_TLF-Normal comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Changed: **Menubar-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1465230800280-b57bd782e7e1-075f70c0c924/Menubar-Normal-b57bd782e7e1.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1465230800280-b57bd782e7e1-075f70c0c924/Menubar-Normal-075f70c0c924.png)  
   Menubar-Normal comparison
 * Changed: **Folders_TLF-Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1465230800280-b57bd782e7e1-075f70c0c924/Folders_TLF-Normal-b57bd782e7e1.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1465230800280-b57bd782e7e1-075f70c0c924/Folders_TLF-Normal-075f70c0c924.png)  
   Folders_TLF-Normal comparison

**Comment by MarcoPolo** [comment]
Do you want another review on this or is this good?

**Comment by chrisnojima** [comment]
i have a thumb so i assume its good. i'll merge this 2moro

**Comment by CumHoleZH** [comment]
Comment

**Pull request merged**